Apps/Examples/Examples/All Examples/Custom2DPuckExample.swift

```diff
@@ -50,37 +175,79 @@ public class Custom2DPuckExample: UIViewController, ExampleProtocol {
+        button.setTitleColor(.white, for: .normal)
+        button.backgroundColor = #colorLiteral(red: 0, green: 0.4784313725, blue: 0.9882352941, alpha: 1)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.layer.cornerRadius = 20
+        button.addTarget(self, action: #selector(changePuckOptions(sender:)), for: .touchUpInside)
+        view.addSubview(button)
+
+        // Set button location
+        NSLayoutConstraint.activate([
+            button.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -24),
+            button.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            button.widthAnchor.constraint(equalToConstant: 200),
+            button.heightAnchor.constraint(equalToConstant: 40)
+        ])
     }
 
-    func syncPuckAndButton() {
-        // Update puck config
-        var configuration = Puck2DConfiguration(topImage: UIImage(named: "star"))
-        configuration.showsAccuracyRing = showsAccuracyRing
-        configuration.accuracyRingColor = UIColor.skyBlue
-        configuration.accuracyRingBorderColor = UIColor.lightGray
+    @objc public func changePuckOptions(sender: UIButton) {
+        let alert = UIAlertController(title: "Toggle Puck Options",
+                                      message: "Select an options to toggle.",
+                                      preferredStyle: .actionSheet)
+
+        alert.addAction(UIAlertAction(title: "Toggle Puck visibility", style: .default, handler: { _ in
+            self.showsPuck.toggle()
+        }))
+
+        alert.addAction(UIAlertAction(title: "Toggle Puck image", style: .default, handler: { _ in
+            self.puckImage.toggle()
+        }))
+
+        alert.addAction(UIAlertAction(title: "Toggle bearing visibility", style: .default, handler: { _ in
+            self.showsBearing.toggle()
+        }))
+
+        alert.addAction(UIAlertAction(title: "Toggle accuracy ring", style: .default, handler: { _ in
+            self.showsAccuracyRing.toggle()
+        }))
+
+        alert.addAction(UIAlertAction(title: "Toggle bearing source", style: .default, handler: { _ in
+            self.bearingSource.toggle()
+        }))
 
-        mapView.location.options.puckType = .puck2D(configuration)
+        alert.addAction(UIAlertAction(title: "Toggle Map Style", style: .default, handler: { _ in
+            self.style.toggle()
+        }))
 
-        // Update button title
-        let title: String = showsAccuracyRing ? "Disable Accuracy Radius" : "Enable Accuracy Radius"
-        toggleAccuracyRadiusButton.setTitle(title, for: .normal)
+        alert.addAction(UIAlertAction(title: "Toggle Projection", style: .default, handler: { _ in
+            self.projection.toggle()
+        }))
+
+        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel))
+
+        present(alert, animated: true)
+    }
+
+    func updatePuckUI() {
+        if showsPuck == .isVisible {
```

Comment:
An idea behind enum-per-state is that you then use them in the `swift-case` statement. It would make a compilation time checker over any new type of state (`case`).